### PR TITLE
Enable pub/sub fanout for StatefulSets by supporting podName in metadata

### PR DESCRIPTION
**Pull request by CacheCoww**
Signed-off-by: CacheCoww <hmsmith2510@gmail.com>

# Description

- Adding metadata option for {podName} to allow for population of the podName as consumerID

## Issue reference

Please reference the issue this PR will close: #4195

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by yaron2** [comment]
@artursouza can you review this?

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -2289,6 +2289,12 @@ func (a *DaprRuntime) convertMetadataItemsToProperties(items []components_v1alph
 		for strings.Contains(val, "{uuid}") {
 			val = strings.Replace(val, "{uuid}", uuid.New().String(), 1)
 		}
+		for strings.Contains(val, "{podName}") {
+			if a.podName == "" {
+				log.Warnf("could not get podname property")
```
Comment (yaron2) [inline pkg/runtime/runtime.go:2294]: Should be `could not get podName property`

**Review by yaron2 — CHANGES_REQUESTED** [review]

**Comment by yaron2** [comment]
@CacheCoww is this ready for review?

**Review by yaron2 — APPROVED** [review]

File: pkg/runtime/runtime.go [hunk 2]
```diff
@@ -2347,6 +2347,12 @@ func (a *DaprRuntime) convertMetadataItemsToProperties(items []components_v1alph
 		for strings.Contains(val, "{uuid}") {
 			val = strings.Replace(val, "{uuid}", uuid.New().String(), 1)
 		}
+		for strings.Contains(val, "{podName}") {
+			if a.podName == "" {
+				log.Warnf("could not get podName property")
```
Comment (artursouza) [inline pkg/runtime/runtime.go:2352]: Please, fail instead. Component should not init at all and give clear error. Warnings can easily be ignored.

**Review by artursouza — CHANGES_REQUESTED** [review]

File: pkg/runtime/runtime.go (hunk 2 above)
Comment (yaron2) [reply pkg/runtime/runtime.go:2349]: Ping @CacheCoww please fix this so we can merge.

**Review by artursouza — APPROVED** [review]

**Review by yaron2 — APPROVED** [review]

**Pull request merged**